src/db/commands.rs

```diff
@@ -417,3 +434,30 @@ fn handle_user_result(
     user.connected_at = connected_at;
     Ok(user)
 }
+
+/// Like Result::ok, convert from Result<T, E> to Option<T> but applying a
+/// function to the Err value
+fn ok_or_inspect<T, E, F>(result: StdResult<T, E>, op: F) -> Option<T>
```

Comment:
almost tempted to make this into a whole trait extension, time will tell how useful it is

Reply:
Well, right now you're using it just to contain a call to `conversion_err`. (I *almost* wonder if it might just be better to combine them into one call to reduce some of the templating, but that's a super minor nit.)

It'll be interesting to see how these play out in the future.

Reply:
I had it that way to start with then ended up here. Already had in mind I wanted an Option::ok_or_else like call for Result (but it doesn't have one because it would be dropping Err in that case).

It had "ua.notification_read.error" either embedded inside of it or that also needed to be an argument, so went for the generic combinator I really wanted